### ISSUE TITLE
Biomod response must be numeric

### ISSUE THREAD

**Issue by AugustT** [zoonproject/zoon#101]
There is a requirement in biomod that the resoponse variable must be numeric (not integer)

The CWBZimbabwe module returns occupancy with value as an integer (0s and 1s), therefore:

```R
library(zoon)

# Works
workflow(occurrence = CWBZimbabwe,
         covariate = Bioclim(extent = c(20, 40, -25, -10)),
         process = NoProcess,
         model = LogisticRegression,
         output = SurfaceMap)

# Doesn't work
workflow(occurrence = CWBZimbabwe,
         covariate = Bioclim(extent = c(20, 40, -25, -10)),
         process = NoProcess,
         model = BiomodModel(modelType = 'GLM'),
         output = SurfaceMap)

df <- tmpZoonWorkflow$process.output[[1]]$df

# Doesn't work
biomodData <- BIOMOD_FormatingData(resp.var = df$value, 
                                   expl.var = df[,6:NCOL(df), drop=FALSE], resp.xy = df[,c('longitude', 'latitude')], 
                                   resp.name = 'Species')
# Does work
biomodData <- BIOMOD_FormatingData(resp.var = as.numeric(df$value), 
                                   expl.var = df[,6:NCOL(df), drop=FALSE], resp.xy = df[,c('longitude', 'latitude')], 
                                   resp.name = 'Species')
```

I think it is perfectly acceptable that the value column is a integer, I have no idea why biomod insists on a numeric. I suggest a simple fix in BiomodModel such as:

```R
if(class(df$value) == 'integer') df$value <- as.numeric(df$value)
```

I think this is preferable to insisting that all occurrence modules return value as a numeric rather than integer.

I would be interested to hear the thoughts of @timcdlucas in case you are aware of good reason for the integer requirement...

**Comment by goldingn** [comment]
I agree, switching to numeric only for biomod seems sensible. I imagine there will be many more similar quirks of other packages.

**Comment by timcdlucas** [comment]
It's because they check the type of [resp.var in a cruddy way.](https://github.com/cran/biomod2/blob/master/R/BIOMOD_FormatingData.R)

    # 1. Checking input params class
    available.types <- c( 'numeric', 'data.frame', 'matrix', 
                          'RasterLayer', 'RasterStack',
                          'SpatialPointsDataFrame', 'SpatialPoints')
    ###### resp.var
    if(!(class(resp.var) %in% available.types)){
      stop( paste("Response variable must be one of ", toString(available.types), sep="") )
    }

So no good reason and should probably be changed but don't know that they accept PRs.

So agree to switch. 

I can't remember if we've discussed it before, but does most of zoon accept factors? Seems a reasonable input.

**Comment by AugustT** [comment]
Fixed in commit `97b8a14d50e0df26f02f36688ead5bf84d7a6b70` to modules
